src/pip/_internal/commands/install.py

```diff
@@ -539,84 +535,122 @@ def _warn_about_conflicts(self, conflict_details):
+        raise CommandError("Different installation locations are implied by "
+                           "{} and {}.".format(rest, last))
+
+    if target_dir:
+        if root_path is not None:
+            target_dir = change_root(root_path, target_dir)
+        if os.path.exists(target_dir) and not os.path.isdir(target_dir):
+            raise InstallationError("Target path exists but is not "
+                                    "a directory: {}".format(target_dir))
+        if not test_writable_dir(target_dir):
             raise InstallationError(
-                "Can not perform a '--user' install. User site-packages "
-                "are not visible in this virtualenv."
-            )
-        logger.debug("User install by explicit request")
-        return True
-
-    # If we are here, user installs have not been explicitly requested/avoided
-    assert use_user_site is None
-
-    # user install incompatible with --prefix/--target
-    if prefix_path or target_dir:
-        logger.debug("Non-user install due to --prefix or --target option")
+                "Cannot write to target directory: {}".format(target_dir))
+        logger.debug("Non-user install due to specified target directory")
         return False
-
-    # If user installs are not enabled, choose a non-user install
-    if not site.ENABLE_USER_SITE:
-        logger.debug("Non-user install because user site-packages disabled")
+    if prefix_path:
+        if not site_packages_writable(root=root_path, isolated=isolated_mode,
+                                      prefix=prefix_path):
+            raise InstallationError(
+                "Cannot write to prefix path: {}".format(prefix_path))
+        logger.debug("Non-user install due to specified prefix path")
         return False
 
-    # If we have permission for a non-user install, do that,
-    # otherwise do a user install.
-    if site_packages_writable(root=root_path, isolated=isolated_mode):
+    # ENABLE_USER_SITE shows user site-packages directory status:
+    # * True means that it is enabled and was added to sys.path.
+    # * False means that it was disabled by user request
+    #   (with -s or PYTHONNOUSERSITE).
+    # * None means it was disabled for security reasons
+    #   (mismatch between user or group id and effective id)
+    #   or by an administrator.
+    if use_user_site is not None:
+        use_user_site = bool(use_user_site)
+        if use_user_site:
+            logger.debug("User install by explicit request")
+        else:
+            logger.debug("Non-user install by explicit request")
+        if use_user_site and ENABLE_USER_SITE is None:
+            raise InstallationError("User site-packages are disabled for "
+                                    "security reasons or by an administrator.")
+    elif ENABLE_USER_SITE is None:
+        logger.debug("Non-user install since user site-packages are disabled "
+                     "for security reasons or by an administrator.")
```

Comment:
Assuming/implying "for security reasons or by an administrator" here is incorrect.

I'd prefer we avoid guessing *why* the variable has a given value and simplify this logic to say:

```suggestion
        logger.debug(
            "User site-packages is disabled. Will use environment's default site-packages."
        )
```

Reply:
This is taken straight from [the docs](https://docs.python.org/3/library/site.html#site.ENABLE_USER_SITE).  Since this is debug level and there's a distinction between what an user can do (when ENABLE_USER_SITE is False) and what perse probably can't (when ENABLE_USER_SITE is None), I think it might be reasonable to keep this information.  I'll rephrase this a little to remove the awkward `Non-user install` thing though.

Reply:
Even so, I'd much rather we log the value of ENABLE_USER_SITE rather than reproducing that information incompletely.

"None means it was disabled for security reasons (mismatch between user or group id and effective id) [snip]" reads very differently from "user site-packages are disabled for security reasons or [snip]".

Reply:
Does 93062547 do what you mean?